src/smtsolvers/SimpSMTSolver.cc

```diff
@@ -180,7 +180,7 @@ lbool SimpSMTSolver::solve_(bool do_simp, bool turn_off_simp)
 //=================================================================================================
 // Added code
 
-bool SimpSMTSolver::addOriginalSMTClause(const vec<Lit> & smt_clause, opensmt::pair<CRef, CRef> & inOutCRefs)
+bool SimpSMTSolver::addOriginalSMTClause(vec<Lit> && smt_clause, opensmt::pair<CRef, CRef> & inOutCRefs)
```

Comment:
So we can move it to `CoreSMTSolver::addOriginalClause_`.

Reply:
Without copies you mean?
Ok on one hand, on the other we lose the guarantee that the vector won't be changed in the function...

Reply:
Yes, without copies.
We do not need such the guarantees.
The change means that we are consuming the vector. Thus, the caller knows that when it passes us the vector, it cannot access it again, because we have consumed it.